robot_smach_states/tutorial/challenge_tutorial.md

```diff
@@ -0,0 +1,236 @@
+# Creating a TU/e RoboCup challenge
+
+This tutorial explains how to setup a new RoboCup challenge. It assumes that you have finished the [ROS tutorials](http://wiki.ros.org/ROS/Tutorials) as well as the [Smach tutorials](http://wiki.ros.org/smach/Tutorials). Furthermore, it assumes that the simulator as well as 'middleware' are running. (ToDo: add links.)
+
+## Setting up the boiler plate
+
+### Setting up the package
+
+* Browse to the `tue_robocup` folder on your machine (ToDo: either create it directly in workspace, create link or whole target meuk):
+```
+cd ~/ros/$ROS_DISTRO/repos/https_/github.com/tue-robotics/tue_robocup.git/
+```
+* Create the package:
+```
+catkin_create_pkg challenge_example robot_skills robot_smach_states
+```
+This creates a ROS package called `challenge_example`. It depends on the `robot_skills`
+and the `robot_smach_states` package. This can be seen when inspecting the `package.xml` that is located inside your newly created `challenge_example` folder.
+* Make and source your `~/.bashrc` to make sure the package is in your `ROS_PACKAGE_PATH`.:
+```
+tue-make
+source ~/.bashrc
+```
+and go to your newly created package:
+```
+roscd challenge_example
+```
+to see if everything has worked.
+
+### Create the ROS node
+
+* In the `challenge_example` package, create a new `scripts` folder and enter it:
+```
+mkdir scripts
+cd scripts
+```
+* Create you new node and make it executable:
+```
+touch challenge_example
+chmod +x challenge_example
+```
+Using `ls` in your terminal you can see that the file is green.
+* Open your file in Pycharm and add the following contents:
+
+```python
+#!/usr/bin/python
+
+# ROS
+import rospy
+
+
+if __name__ == "__main__":
+
+    rospy.init_node("state_machine_example")
+    rospy.loginfo("Hello world")
+```
+
+* Now you can run the challenge:
+```
+rosrun challenge_example challenge_example
+```
+
+### Creating a robot api object
+* Update your example script to:
+
+```python
+#!/usr/bin/python
+
+# ROS
+import rospy
+
+# TU/e Robotics
+from robot_skills.util.robot_constructor import robot_constructor
+
+
+if __name__ == "__main__":
+
+    rospy.init_node("state_machine_test")
+    rospy.loginfo("Hello world")
+    robot = robot_constructor("amigo")
+    raw_input("Press any key to exit")
+
+```
+
```

Comment:
Running this gives: KeyError: '/amigo/skills/torso/joint_names', because parameters are not loaded yet.
Fix it by running amigo-start, close it and run script again

Reply:
See above, but added reminder in b22087a5bb02c65398bcb173ad49805ef501e8d8